impeller/renderer/backend/vulkan/descriptor_pool_vk.h

```diff
@@ -24,29 +20,21 @@ namespace impeller {
 ///
 ///             Encoders create pools as necessary as they have the same
 ///             threading and lifecycle restrictions.
-///
 class DescriptorPoolVK {
  public:
   explicit DescriptorPoolVK(
       const std::weak_ptr<const DeviceHolder>& device_holder);
 
   ~DescriptorPoolVK();
 
-  std::optional<vk::DescriptorSet> AllocateDescriptorSet(
-      const vk::DescriptorSetLayout& layout,
-      size_t command_count);
+  std::vector<vk::DescriptorSet> AllocateDescriptorSets(
```

Comment:
Lets document what an empty vector would mean. Ideally, this would be a vector of optional elements.

Reply:
What would a std::nullopt descriptor set imply?

Reply:
Nevermind. I thought it was a DescriptorSetUnique. Following this up by making it a status or (or just leave it be) is fine.